File: backend/app/schemas/conversations.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class ConversationCreate(BaseModel):
    channel: str = "web"


class MessageOut(BaseModel):
    id: uuid.UUID
    role: str
    content: str
    response_time_ms: Optional[int] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class ConversationOut(BaseModel):
    id: uuid.UUID
    org_id: uuid.UUID
    status: str
    channel: str
    rating: Optional[int] = None
    created_at: datetime
    closed_at: Optional[datetime] = None
    message_count: int = 0

    model_config = {"from_attributes": True}


class ConversationDetail(ConversationOut):
    messages: list[MessageOut] = []


class RateConversation(BaseModel):
    rating: int  # 1-5

File: frontend/src/api/conversations.ts
import api from "./client";

export interface Conversation {
  id: string;
  org_id: string;
  status: "open" | "closed";
  channel: "web" | "whatsapp" | "instagram";
  rating: number | null;
  created_at: string;
  closed_at: string | null;
  message_count: number;
}

export interface Message {
  id: string;
  role: "user" | "ai";
  content: string;
  response_time_ms: number | null;
  created_at: string;
}

export interface ConversationDetail extends Conversation {
  messages: Message[];
}

export async function listConversations(): Promise<Conversation[]> {
  const { data } = await api.get<Conversation[]>("/conversations");
  return data;
}

export async function createConversation(channel = "web"): Promise<Conversation> {
  const { data } = await api.post<Conversation>("/conversations", { channel });
  return data;
}

export async function getConversation(id: string): Promise<ConversationDetail> {
  const { data } = await api.get<ConversationDetail>(`/conversations/${id}`);
  return data;
}

export async function closeConversation(id: string): Promise<Conversation> {
  const { data } = await api.patch<Conversation>(`/conversations/${id}/close`);
  return data;
}

export async function rateConversation(id: string, rating: number): Promise<Conversation> {
  const { data } = await api.patch<Conversation>(`/conversations/${id}/rate`, { rating });
  return data;
}
